### PR TITLE
Export RazorDocumentManager using roslyn services

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Fixes https://github.com/aspnet/Razor/issues/1799

FYI @mkArtakMSFT

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManager.cs [hunk 1]
```diff
@@ -3,27 +3,23 @@
...
         private readonly ForegroundDispatcher _foregroundDispatcher;
         private readonly RazorEditorFactoryService _editorFactoryService;
         private readonly TextBufferProjectService _projectService;
 
-        [ImportingConstructor]
         public DefaultRazorDocumentManager(
             RazorEditorFactoryService editorFactoryService,
             TextBufferProjectService projectService,
-            VisualStudioWorkspaceAccessor workspaceAccessor)
+            ForegroundDispatcher dispatcher)
```
Comment (NTaylorMullen) [inline src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManager.cs]: put dispatcher first for consistency across our types. You can utilize the refactor method dialog to do this so you don't have to change all references manually.

File: src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManagerFactory.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    [ExportLanguageServiceFactory(typeof(RazorDocumentManager), RazorLanguage.Name, ServiceLayer.Default)]
+    internal class DefaultRazorDocumentManagerFactory : ILanguageServiceFactory
+    {
+        private readonly RazorEditorFactoryService _editorFactoryService;
+        private readonly TextBufferProjectService _projectService;
+
+        [ImportingConstructor]
+        public DefaultRazorDocumentManagerFactory(
+            RazorEditorFactoryService editorFactoryService,
+            TextBufferProjectService projectService)
```
Comment (NTaylorMullen) [inline src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManagerFactory.cs]: This type should also come from Roslyn services. Can be a follow up.

File: src/Microsoft.VisualStudio.Editor.Razor/RazorDocumentManager.cs [hunk 3]
```diff
@@ -2,12 +2,13 @@
...
 
 using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Host;
 using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Editor;
 
 namespace Microsoft.VisualStudio.Editor.Razor
 {
-    internal abstract class RazorDocumentManager
+    internal abstract class RazorDocumentManager : ILanguageService
```
Comment (NTaylorMullen) [inline src/Microsoft.VisualStudio.Editor.Razor/RazorDocumentManager.cs]: Shower Thought: A general rule of thumb with the approach we're taking in Razor is, if we have an `internal abstract class` then it should always be an `ILanguageService` 😄

File: src/Microsoft.VisualStudio.Editor.Razor/RazorTextViewConnectionListener.cs [hunk 4]
```diff
@@ -20,22 +20,17 @@ internal class RazorTextViewConnectionListener : ITextViewConnectionListener
         private readonly RazorDocumentManager _documentManager;
 
         [ImportingConstructor]
-        public RazorTextViewConnectionListener(
-            VisualStudioWorkspaceAccessor workspaceAccessor,
-            RazorDocumentManager documentManager)
+        public RazorTextViewConnectionListener(VisualStudioWorkspaceAccessor workspaceAccessor)
```
Comment (NTaylorMullen) [inline src/Microsoft.VisualStudio.Editor.Razor/RazorTextViewConnectionListener.cs]: Beautiful 👍

File: src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManagerFactory.cs (hunk 2 above)
Comment (ajaybhargavb) [reply src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManagerFactory.cs]: Added it to the checklist https://github.com/aspnet/Razor/issues/1799#issue-279962189

File: src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManager.cs (hunk 1 above)
Comment (ajaybhargavb) [reply src/Microsoft.VisualStudio.Editor.Razor/DefaultRazorDocumentManager.cs]: Done

**Pull request merged**